RELEASE_PROCESS.md

```diff
@@ -0,0 +1,36 @@
+# Building a Compose release
```

Comment:
Thanks for this.  Good work!  I think this could be clarified a bit around branch naming and also around the docs release. Also, point folks back to contributing so they don't try to use this build procedure for building dev branches.

BTW, didn't have time to test or proof so ware the typos!

---> https://gist.github.com/moxiegirl/51579a1e811227fa05b2

# Build and release Compose

This section explains how to build and release Compose for distribution. If you are building binaries for developing contributions, see the [contributing page}(CONTRIBUTING.MD) instead. 


## Building binaries for Linux

Use `script/build-linux` to build the Linux binary inside a Docker container:

    $ script/build-linux
    
## Build the binary for OS x

For official releases, you should build inside a Mountain Lion VM for proper
compatibility. This process relies on Homebrew (`brew`) to install Python and set up a virtual environment. If you don't have Homebrew installed, the prepare script installs it for you.  

1. Prepare your environment by running:

        $ script/prepare-osx

2. Build the Mac OS X binary inside a virtualenv:

        $ script/build-osx    

## Release process

1. On your fork, create a `bump-X.Y.Z` branch against the `upstream/release` branch.

  For example: 
  
       $ checkout -b  bump-1.7.0 upstream/Release

2. In your bump branch:
 - Update the version in the `compose/__init__.py` file.
 - Update the binary URL in the `docs/install.md` file.
 - Add  release notes to the `CHANGES.md` file.
3. Open pull request back to the upstream `release` branch.
4. Create an unpublished GitHub release with release notes
5. Build Linux version on any Docker host with the `script/build-linux` script.
6. Build OS X version on Mountain Lion with the `script/build-osx`script.
7. Attach the build output to the release as
   `docker-compose-Linux-x86_64` and `docker-compose-Darwin-x86_64`
8. Overwrite the `docs` branch with the `release` content.

        $ git checkout -b docs release || git checkout docs
        $ git fetch
        $ git reset --hard origin/release
        $ git push -f origin docs

9. Publish GitHub release to create the tag.
10. Upload PyPi package

        $ git checkout $VERSION
        $ python setup.py sdist upload